src/wasm/wasm-type.cpp

```diff
@@ -310,6 +318,20 @@ bool isTemp(HeapType type) {
   return !type.isBasic() && getHeapTypeInfo(type)->isTemp;
 }
 
+Type asCanonical(Type type) {
```

Comment:
this also seems worth a comment. trying to figure out from the code (and from getCanonical at the same time), I'm still unsure.